libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/LPC17xx_LPC40x8/NetworkInterface.c

```diff
@@ -0,0 +1,806 @@
+{
+   BaseType_t x;
+
+   for (x = 0; x < ipconfigNUM_NETWORK_BUFFER_DESCRIPTORS; x++)
+   {
+      /* pucEthernetBuffer is set to point ipBUFFER_PADDING bytes in from the
+         beginning of the allocated buffer. */
+      pxNetworkBuffers[x].pucEthernetBuffer = &(ucBuffers[x][ipBUFFER_PADDING]);
+
+      /* The following line is also required, but will not be required in
+         future versions. */
+      *((uint32_t *)&ucBuffers[x][0]) = (uint32_t)&(pxNetworkBuffers[x]);
+   }
+}
+#endif
+
+/* See https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/Embedded_Ethernet_Porting.html
+ *
+ * xNetworkInterfaceInitialise() must prepare the Ethernet MAC to send and receive data.
+ * In most cases this will just involve calling whichever initialise function is provided with
+ * the Ethernet MAC peripheral drivers - which will in turn ensure the MAC hardware is enabled
+ * and clocked, as well as configure the MAC peripheral's DMA descriptors.
+ *
+ * xNetworkInterfaceInitialise() does not take any parameters, returns pdPASS if the initialisation was successful,
+ * and returns pdFAIL if the initialisation fails.
+ */
+BaseType_t xNetworkInterfaceInitialise(void)
+{
+   BaseType_t xReturn = pdFAIL;
+   static bool RxTaskCreated = pdFALSE; /* Guard against creating more than one receive task */
+   static bool HwInitialised = pdFALSE;
+   BaseType_t LinkStatus = 0;
+   BaseType_t RxFilterControl = 0;
+   TickType_t xLastWakeTime = xTaskGetTickCount(); /* Initialise with the current time. */
+
+   if (HwInitialised == pdFALSE)
+   {
+#if ( iplpcINIT_PORT_PINS == 1 )
+      /* Start by setting up the pins for the desired MAC-PHY interface type. */
+      prvBoard_ENET_Init(configUSE_RMII);
+#endif // ( iplpcINIT_PORT_PINS == 1 )
+
+      /* Initialise the peripheral using MII or RMII according to configuration */
+      Chip_ENET_Init(LPC_ETHERNET, iplpcUSE_RMII);
+
+      /* Initialise the PHY clock for 2.5MHz and address */
+      Chip_ENET_SetupMII(LPC_ETHERNET,
+                         Chip_ENET_FindMIIDiv(LPC_ETHERNET, 2500000),
+                         ENET_PHYDEF_PHYADDR);
+
+      /* Initialise the external PHY */
+      if (xPhyConfigure() == pdPASS)
+      {
+         /* Save MAC address */
+         Chip_ENET_SetADDR(LPC_ETHERNET, ucMACAddress);
+
+         /* Clear all MAC address hash entries. */
+         LPC_ETHERNET->RXFILTER.HashFilterH = 0;
+         LPC_ETHERNET->RXFILTER.HashFilterL = 0;
+
+         /* Add the station MAC address to the filter hash table */
+         prvAddMACAddress(ucMACAddress);
+
+#if (ipconfigUSE_LLMNR == 1)
+         /* Add the LLMNR MAC address to the filter hash table */
+         prvAddMACAddress (xLLMNR_MACAddress);
+#endif /* ipconfigUSE_LLMNR == 1 */
+
+         /* Enable packet reception according to configuration requirements. */
+         // AWM - for testing, we'll accept anything. TODO sort out config #defines
```

Comment:
Is the code on lines 617-620 configuring the RX filter to receive all frames as the _"...we'll accept anything."_ comment seems to indicate?  If so, that would only be suitable for a testing environment in most applications so I'd recommend checking that you're only configuring to receive on the station MAC, LLMNR multicast address, and the broadcast address.